Sources/AzureIoTMiddlewareForSwift/AzureIoT.swift

```diff
@@ -0,0 +1,318 @@
+import Foundation
+import MQTT
+import NIOSSL
+import Crypto
+import CAzureSDKForCSwift
+import AzureIoTUniversalMiddleware
+
+extension StringProtocol {
+    subscript(offset: Int) -> Character {
+        self[index(startIndex, offsetBy: offset)]
+    }
+}
+
+public class AzureIoT: MQTTClientDelegate {
+    // Section 1: Variables
+    private var azureIotConfig: azure_iot_config_t = azure_iot_config_t()
+    private var internalClient: azure_iot_t = azure_iot_t()
+    private var internalBuffer: AzSpan = AzSpan(size: 2048);
+
+    private var MQTT_PASSWORD_LIFETIME_IN_MINUTES: UInt32 = 60
+
+    private var userAgent: AzSpan = AzSpan(text: "c%2F1.3.0-beta.1(swift)")
```

Comment:
currently the swift lib uses [AzSpan(text: "azsdk-c%2Fswift%2F\(AZ_SDK_VERSION_STRING)")](https://github.com/Azure-Samples/azure-sdk-for-c-swift/blob/dc14ff070fb79434bce5bae73d3e6e12b4f5e5fb/Sources/CAzureSDKForCSwift/AzureIoTClient.swift#L277)

Should we use that but tweak a little?
```suggestion
    private var userAgent: AzSpan = AzSpan(text: "azsdk-c%2Fswift-mid%2F\(AZ_SDK_VERSION_STRING)")
```

adding the `-mid` to distinguish